File: src/main/java/com/fury/game/entity/combat/scripts/slayer/NechryaelCombat.java
package com.fury.game.entity.combat.scripts.slayer;

import com.fury.core.model.node.entity.actor.figure.Figure;
import com.fury.core.model.node.entity.actor.figure.mob.Mob;
import com.fury.game.entity.combat.CombatScript;
import com.fury.game.npc.slayer.Nechryael;
import com.fury.game.world.update.flag.block.Animation;
import com.fury.util.Misc;

public class NechryaelCombat extends CombatScript {

    @Override
    public Object[] getKeys() {
        return new Object[] {"Nechryael"};
    }

    @Override
    public int attack(Mob mob, Figure target) {
        if(mob instanceof Nechryael) {
            Nechryael n = (Nechryael) mob;
            if (Misc.random(10) == 0 && !n.hasActiveSpawns())
                n.summonDeathSpawns();
        }
        mob.perform(new Animation(mob.getCombatDefinition().getAttackAnim()));
        delayHit(mob, 0, target, getMagicHit(mob, getRandomMaxHit(mob, mob.getCombatDefinition().getAttackStyle(), target)));
        return mob.getCombat().getAttackSpeed();
    }
}

File: src/main/kotlin/com/fury/game/node/entity/mob/combat/NechryaelCombat.kt
package com.fury.game.node.entity.mob.combat

import com.fury.core.model.node.entity.actor.figure.Figure
import com.fury.game.node.entity.actor.figure.mob.MobCombat
import com.fury.game.npc.slayer.Nechryael

class NechryaelCombat(private val mob: Nechryael) : MobCombat(mob) {
    override fun sendDeath(source: Figure?) {
        super.sendDeath(source)
        if (mob.hasActiveSpawns())
            mob.removeDeathSpawns()
    }
}